node/node.go

```diff
@@ -696,6 +696,18 @@ func (n *nodeImpl) OnStart() error {
 			n.eventBus.Logger.Error("failed to emit the statesync start event", "err", err)
 		}
 
+		if n.config.P2P.UseLegacy {
+			pl := n.sw.Peers().List()
+			peers := make([]types.NodeID, 0, len(pl))
+			for _, p := range pl {
+				peers = append(peers, p.ID())
+			}
+
+			n.stateSyncReactor.InitPeers(peers)
```

Comment:
I'm assuming we can't currently do this in the reactor constructor? That would definitely be my preference, but it seems possible the way we start `Service`s up makes that impossible.

Reply:
yep, I agree with you. That's sort of the whole reason for the Node package, after a fashion. 

It's sort of further complicated by the fact that once the router/switch start they can start connecting to peers, but the reactor hasn't started until somewhat later, and *this* happens right before we start the state sync process.